connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfigTransformer.java

```diff
@@ -38,10 +38,16 @@ public WorkerConfigTransformer(Worker worker, Map<String, ConfigProvider> config
         this.configTransformer = new ConfigTransformer(configProviders);
     }
 
+    public Map<String, String> transform(Map<String, String> configs) {
+        return transform(null, configs);
```

Comment:
Why should we have `null` connector `transform`?

Reply:
There is no connector name when `validateConnectorConfigs` is called here: https://github.com/apache/kafka/blob/trunk/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorPluginsResource.java#L81 .   Also, the connector name is only used to schedule a reload of the config, which doesn't apply during validation.